code/script.js

```diff
@@ -227,99 +229,157 @@ const setSecret = () => {
   } else if (category === 'other') {
-    // Similar to the one above
-  } else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      alert (
+        `Yes, the person is ${value}! Keep all people that is ${value}.`
+        )
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(
+        `No, the person isn't ${value}! Remove all people that are ${value}.`
+        )
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
     }
-  }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
 
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+  } else if (category === 'hair') {
+    if (keep) {
+      alert (
+        `Yes, the person has ${value} hair! Keep all people that has ${value} hair.`
+        )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+
+    } else {
+      alert (
+        `No, the person doesn't has ${value} hair! Remove all people that has ${value} hair.`
+        )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+      }
+  
 
-  // Invoke a function to redraw the board with the remaining people.
+  } else if (category === 'eyes') {
+    if (keep) {
+        alert (
+          `Yes, the person has ${value} eyes! Keep all people with ${value} eyes.`
+          )
+        charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+
+      } else {
+        alert (
+          `No, the person doesn't have ${value} eyes! Remove all people with ${value} eyes.`
+          )
+        charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+      }
+    }
+
+  // This function renders the board when som characters are removed.
+  generateBoard()
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
+//this function confirms and stores the guess
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  const confirmedGuess = confirm(
+    `Are you really sure you want to take a guess on ${personToConfirm}?`
+    );
+
+    if (confirmedGuess) {
```

Comment:
Would have been fun to have an alert if the person playing chickens out maybe? But that's just a suggestion.